src/main/scala/za/co/absa/commons/s3/S3Location.scala

```diff
@@ -30,29 +30,76 @@ trait S3Location {
+   * These can be helpful to quickly get desired location as a string.
+   *
+   * @return @return formatted s3 string with concrete protocol
+   */
+  def asS3LocationString: String = s"s3://$bucketName/$path"
+  def asS3ALocationString: String = s"s3a://$bucketName/$path"
 }
 
 object SimpleS3Location {
 
   /**
    * Generally usable regex for validating S3 path, e.g. `s3://my-cool-bucket1/path/to/file/on/s3.txt`
+   * S3 path can be represented by a S3 bucket name or S3 Access Point Alias that can be used as a reference to a bucket.
    * Protocols `s3`, `s3n`, and `s3a` are allowed.
-   * Bucket naming rules defined at [[https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules]] are instilled.
+   *
+   * - Bucket naming rules are defined at
+   *    [[https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules]]
+   * - S3 Access Point naming rules are defined at
+   *    [[https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-access-points.html]]
+   *
+   * Note: this regexp is not 100% strict and exact, according to the AWS S3 naming rules above. That would require
+   *   something more powerful than just a regexp. Some differences worth mentioning:
+   *
+   *   - The dot character is allowed in S3 bucket name, but not in S3 Access Point. The use of dots in bucket name
+   *     is rare, but we decided to keep it in the regexp - so that we have a pattern that satisfies both,
+   *     access point and bucket name, but it's not as strict as it could be.
+   *
+   *   - Maximum length of S3 Access Point is 50, and max length of S3 bucket name is 63. We've
+   *     decided to ignore this detail and use 63.
+   *
+   *   - No check for first and last character (or prefix/suffix) in a bucket name.
    */
   private val S3LocationRx: Regex = "^(s3[an]?)://([-a-z0-9.]{3,63})/(.*)$".r
 
+  def apply(path: String): SimpleS3Location = {
+    path.toSimpleS3Location.getOrElse(throw new IllegalArgumentException(s"Could not parse S3 location from $path!"))
+  }
+
   implicit class SimpleS3LocationExt(val path: String) extends AnyVal {
 
     def toSimpleS3Location: Option[SimpleS3Location] = PartialFunction.condOpt(path) {
-      case S3LocationRx(protocol, bucketName, relativePath) =>
-        SimpleS3Location(protocol, bucketName, relativePath)
+      case S3LocationRx(protocol, bucketOrAccessPointAlias, relativePath) =>
+        SimpleS3Location(protocol, bucketOrAccessPointAlias, relativePath)
     }
 
     def isValidS3Path: Boolean = S3LocationRx.pattern.matcher(path).matches
-  }
 
-  def apply(path: String): SimpleS3Location = {
-    path.toSimpleS3Location.getOrElse(throw new IllegalArgumentException(s"Could not parse S3 location from $path!"))
+    def withoutTrailSlash: SimpleS3Location = {
+      val parsedS3Location = SimpleS3Location(path)
+
+      if (parsedS3Location.path.endsWith("/")) parsedS3Location.copy(path=parsedS3Location.path.dropRight(1))
```

Comment:
Do you / we have some coding standards somewhere, documented? I have no problems with it but when I used Reformat Code feature in IntelliJ, nothing happened (and I usually used no spacing around `=` ). But I don't have a strong preference around this so I'm gonna make the changes

Reply:
https://docs.scala-lang.org/style/

Specifically:

### Method Invocation
> https://docs.scala-lang.org/style/method-invocation.html
> Named parameters in a method invocation should be treated as regular parameters (spaced accordingly following the comma) with a space on either side of the equals sign:
> ```scala
> foo(x = 6, y = 7)
> ```

### The curly braces on control statements
Scala code style guide doesn't explicitly suggest the use of curly braces on one-line statements if the other part of the expression is multi-line, but it tells this:
> https://docs.scala-lang.org/style/control-structures.html#curly-braces
> Curly-braces should be omitted in cases where the control structure represents a pure-functional operation and all branches of the control structure (relevant to `if/else`) are single-line expressions.

Another example could be the following paragraph from the Databricks code style guide:
> https://github.com/databricks/scala-style-guide#curly-braces
> Put curly braces even around one-line conditional or loop statements. The only exception is if you are using if/else as an one-line ternary operator that is also side-effect free.
> ```scala
> // Correct:
> if (true) {
>   println("Wow!")
> }
> 
> // Correct:
> if (true) statement1 else statement2
> 
> // Correct:
> try {
>   foo()
> } catch {
>   ...
> }
> 
> // Wrong:
> if (true)
>   println("Wow!")
> 
> // Wrong:
> try foo() catch {
>   ...
> }
> ```

Both quotes imply consistency - either both branches are inside curly braces or none.

### String interpolation
This is in the ABSA OSS guideline which I do not know where it is currently located :)
But there are number of reasons why to avoid using '+' for string concatenation. Among them are semantical safeness, conciseness and performance (lack of slowness to be more precise, as there used to be issues in the past that should be all solved now, so it should be very comparative to Java native '+' perf).
Sources:
- https://netvl.github.io/scala-guidelines/language-features/string-interpolation.html
- https://nrinaudo.github.io/scala-best-practices/tricky_behaviours/string_concatenation.html